lib/libc/riscv/gen/_setjmp.S

```diff
@@ -39,7 +39,48 @@ __FBSDID("$FreeBSD$");
 
 ENTRY(_setjmp)
 #ifdef __CHERI_PURE_CAPABILITY__
-#pragma message("TODO: RISCV-PURECAP implement _setjmp")
+	/* Store the magic value and stack pointer */
+	cllc	ct0, .Lmagic
+	cld	t0, (ct0)
```

Comment:
You could write this more efficiently as:
```
1:  auipc ct0, %pcrel_hi(.Lmagic)
    ld t0, %pcrel_lo(1b)(ct0)
```

This is the capability equivalent of the `ld reg, sym` pseudo; I wonder if we also want to add purecap variants of those. Probably don't want to add the stores as that's _extremely_ niche (i.e. only really early in boot, nothing more), but the loads are useful for this...

Reply:
I've thought about asking for that for `clgc / cjalr` but it's not that bad to write it out really.

Reply:
Mmm, I missed the one-less instruction part of this before.  It might indeed be nice to have the load pseudo's, though you still have the issue of when to use `cllc` vs `clgc`?  Not sure how the `ld` pseudos in base RISC-V deal with that.

I also think it might be nice to have `call` and `tail` pseudo's, though perhaps we could make them always use `clgc` and add `lcall` and `ltail` pseudos that use `cllc`?  For call and tail it is perhaps less important since `cjalr` doesn't take an immediate so you can't skip an instruction the way a `ld` pseudo would for the above.  It would be nice to avoid having to use explicit `%pcrel_*`